qa-tests-backend/src/test/groovy/ExternalNetworkSourcesTest.groovy

```diff
@@ -192,43 +172,33 @@ class ExternalNetworkSourcesTest extends BaseSpecification {
 
         String supernetName = generateNameWithPrefix("external-source-supernet")
         NetworkEntity supernet = createNetworkEntityExternalSource(supernetName, supernetCIDR)
-        String supernetID = supernet?.getInfo()?.getId()
-        assert supernetID != null
+        String supernetID = supernet.getInfo().getId()
 
         log.info "Verify edge exists from deployment to supernet external source"
-        withRetry(4, 30) {
-            assert NetworkGraphUtil.checkForEdge(deploymentUid, supernetID)
-        }
+        verifyEdge(deploymentUid, supernetID)
 
         log.info "Add smaller subnet subnet external source"
         String subnetName = generateNameWithPrefix("external-source-subnet")
         NetworkEntity subnet = createNetworkEntityExternalSource(subnetName, subnetCIDR)
-        String subnetID = subnet?.getInfo()?.getId()
-        assert subnetID != null
+        String subnetID = subnet.getInfo().getId()
 
         then:
         "Verify edge exists from deployment to subnet external source"
-        withRetry(4, 30) {
-            assert NetworkGraphUtil.checkForEdge(deploymentUid, subnetID, null, 180)
-        }
+        verifyEdge(deploymentUid, subnetID)
 
         and:
         "Verify edge from deployment to supernet exists in older network graph"
-        withRetry(4, 30) {
-            assert NetworkGraphUtil.checkForEdge(
-                    deploymentUid,
-                    supernetID,
-                    Timestamp.newBuilder().setSeconds(System.currentTimeSeconds() - 60*60).build())
-        }
+        verifyEdge(
+                deploymentUid,
+                supernetID,
+                60 * 60)
 
         and:
         "Verify no edge from deployment to supernet exists in recent network graph"
-        withRetry(4, 30) {
-            assert verifyNoEdge(
-                    deploymentUid,
-                    supernetID,
-                    Timestamp.newBuilder().setSeconds(System.currentTimeSeconds() - 60).build())
-        }
+        verifyNoEdge(
+                deploymentUid,
+                supernetID,
+                60)
```

Comment:
My experiments showed that we may need to wait/retry longer for this assertion. How can we define the number of attempts and the pause time using the notation proposed in this PR?

Reply:
currently we have `@Retry(attempts = 4, delay = 30)` we can increase that if needed

Reply:
Can we increase that only for this particular call?

Reply:
Fixed